src/services/codefixes/fixRemoveAbstractModifierInNonAbstractClass.ts

```diff
@@ -0,0 +1,40 @@
+/* @internal */
+namespace ts.codefix {
+    registerCodeFix({
+        errorCodes: [Diagnostics.Abstract_methods_can_only_appear_within_an_abstract_class.code],
+        getCodeActions: (context: CodeFixContext) => {
+            const sourceFile = context.sourceFile;
+            const start = context.span.start;
+            const token = getTokenAtPosition(sourceFile, start);
+
+            Debug.assert(token.kind === SyntaxKind.AbstractKeyword);
+
+            const propertyDeclaration = <PropertyDeclaration>token.parent;
+            const classDeclaration = <ClassDeclaration>propertyDeclaration.parent;
+            const classKeywordStart = classDeclaration.getChildren()[0].getStart();

```

Comment:
This assumes the first child is always the `class` keyword. Does it hold when the class has decorators / modifiers?